playbooks/sample-sap-hypervisor-redhat-ocp-virt-preconfigure.yml

```diff
@@ -6,91 +6,26 @@
     sap_hypervisor_node_preconfigure_platform: redhat_ocp_virt
   tasks:
 
-    - name: Use kubeconfig file specified in environment variable K8S_AUTH_KUBECONFIG if sap_hypervisor_node_preconfigure_ocp_kubeconfig_path is not defined
+    - name: Use kubeconfig file specified in environment variable K8S_AUTH_KUBECONFIG | KUBECONFIG if sap_hypervisor_node_preconfigure_ocp_kubeconfig_path is not defined.
       when: >
-        sap_hypervisor_node_preconfigure_ocp_kubeconfig_path is not defined or
-        sap_hypervisor_node_preconfigure_ocp_kubeconfig_path == None or
-        sap_hypervisor_node_preconfigure_ocp_kubeconfig_path == ''
+        sap_hypervisor_node_preconfigure_kubeconfig is not defined or
+        sap_hypervisor_node_preconfigure_kubeconfig == None or
+        sap_hypervisor_node_preconfigure_kubeconfig == ''
       ansible.builtin.set_fact:
-        sap_hypervisor_node_preconfigure_ocp_kubeconfig_path: "{{ lookup('env', 'K8S_AUTH_KUBECONFIG') | default(None) }}"
+        sap_hypervisor_node_preconfigure_kubeconfig: "{{ lookup('env', 'K8S_AUTH_KUBECONFIG') | lookup('env', 'KUBECONFIG') | default(None) }}"
```

Comment:
"{{ lookup('env', 'K8S_AUTH_KUBECONFIG') | lookup('env', 'KUBECONFIG') | default(None) }}"
-- here we are sending output of lookup('env', 'K8S_AUTH_KUBECONFIG')  to lookup('env', 'KUBECONFIG')
Also is it better to use lookup('env', 'K8S_AUTH_KUBECONFIG')  or lookup('env', 'KUBECONFIG')  -?
Ideally we want flow to be 
1. look for K8S_AUTH_KUBECONFIG, if found stop.
2. if not found look for KUBECONFIG, if found stop
3. if none is there, do we want to fail ? in current case it is silently setting to None which has no use?

Reply:
Hi @geetikakay, thanks for sharing your thoughts on this.

What should be happening is:

1. sap_hypervisor_node_preconfigure_kubeconfig is set --> use this and continue
2. Else look for K8S_AUTH_KUBECONFIG and if set --> use this and continue
3. Else look for KUBECONFIG and if set --> use this and continue
4. Fail with Error message that kubeconfig is not set.

I'll add a task which will handle 4.) which is currently missing.